lib/domain/usecase/room/chat_get_pinned_events_interactor.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:async';
+
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/room/chat_get_pinned_events_state.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatGetPinnedEventsInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Room room,
+  }) async* {
+    Logs().d(
+      "ChatGetPinnedEventsInteractor()::execute()::roomId: ${room.id}",
+    );
+    yield Right(ChatGetPinnedEventsLoading());
+    try {
+      final pinnedEvents = room.pinnedEventIds;
+      final result = await Future.wait(
+        pinnedEvents.map(room.getEventById),
+      );
+
+      if (result.isEmpty) {
+        yield Left(ChatGetPinnedEventsNoResult());
+        return;
+      } else {
+        Logs().d(
+          "ChatGetPinnedEventsInteractor()::execute()::result: ${result.length}",
+        );
+        result.sort((currentEvent, nextEvent) {
```

Comment:
Do we want to sort them in event date time and not pinned date time ?
On other applications such as discord it's sort by when it's has been pinned

Reply:
compare to `telegram`. Tele sort it by the time of event.